middleware/rate_limiter.go

```diff
@@ -0,0 +1,97 @@
+package middleware
+
+import (
+	"github.com/labstack/echo/v4"
+	"golang.org/x/time/rate"
+	"net/http"
+	"sync"
+)
+
+// TokenStore is the interface to be implemented by custom stores.
+type TokenStore interface {
+	// Stores for the rate limiter have to implement the ShouldAllow method
+	ShouldAllow(identifier string) bool
```

Comment:
For the interface the name `TokenStore` seems misleading. What about a generic `RateLimiterStore` or the ideomatic Go `LimiterStorer` interface?

Concerning the interface. `ShouldAllow` could be simplified to `Allow` or `IsAllowed`.
Should we also allow to retrieve the current Limiter for an identifier to be able to report what/why it is above limit?

```go
Allow(identifier string) bool
Status(identifier string) *rate.Limiter     // choose a better name 
```

Reply:
`rate.Limiter` is the engine behind the `InMemoryStore` and is actually not a dependency of the RateLimiter middleware. 
I left the status implementation out of the `TokenStore` because I felt the burden of status checks should be on the store implemented by the user to keep this flexible.

I'd change the interface name to `RateLimiterStore` and the method name to `Allow` thanks for pointing that out